imagetest/test_suites/storageperf/storage_perf_utils.go

```diff
@@ -207,3 +214,113 @@ func checkZypperTransientError(err error, stdout, stderr string) error {
 	}
 	return err
 }
+
+// use the guest attribute to check if hyperdisk is being used. If the guest attribute was not set, assume by default that hyperdisk fio options are not being used.
+func isUsingHyperdisk(ctx context.Context) bool {
+	diskType, err := utils.GetMetadata(ctx, "instance", "attributes", diskTypeAttribute)
+	if err != nil {
+		return false
+	}
+	if diskType == imagetest.HyperdiskExtreme || diskType == imagetest.HyperdiskThroughput || diskType == imagetest.HyperdiskBalanced {
+		return true
+	}
+
+	return false
+}
+
+// function to get num numa nodes
+func getNumNumaNodes() (int, error) {
+	lscpuOut, err := exec.Command("lscpu").CombinedOutput()
+	if err != nil {
+		return 0, err
+	}
+	lscpuOutString := string(lscpuOut)
+	numNumaNodes := -1
+	for _, line := range strings.Split(lscpuOutString, "\n") {
+		lowercaseLine := strings.ToLower(line)
+		if strings.Contains(lowercaseLine, "numa node") {
+			// the last token in the line should be the number of numa nodes
+			tokens := strings.Fields(lowercaseLine)
+			numNumaNodesString := strings.TrimSpace(tokens[len(tokens)-1])
+			i, err := strconv.Atoi(numNumaNodesString)
+			if err == nil {
+				numNumaNodes = i
+				break
+			}
+		}
+	}
+	if numNumaNodes < 0 {
+		return 0, fmt.Errorf("did not find any line with numNumaNodes in lscpu output: %s", lscpuOutString)
+	}
+	return numNumaNodes, nil
+}
+
+// function to get cpu mapping as strings if there is only one numa node
+// returned format is queue_1_cpus, queue_2_cpus, error
+func getCPUNvmeMapping(symlinkRealPath string) (string, string, error) {
+	cpuListCmd := exec.Command("cat", "/sys/class/block/"+symlinkRealPath+"/mq/*/cpu_list")
+	cpuListBytes, err := cpuListCmd.CombinedOutput()
+	if err != nil {
+		return "", "", err
+	}
+	cpuListString := string(cpuListBytes)
+	cpuListOutLines := strings.Split(string(cpuListString), "\n")
+	if len(cpuListOutLines) < 2 {
+		return "", "", fmt.Errorf("expected at least two lines for cpu queue mapping, got string %s with %d lines", cpuListString, len(cpuListOutLines))
+	}
+	queue1Cpus := strings.TrimSpace(cpuListOutLines[0])
+	queue2Cpus := strings.TrimSpace(cpuListOutLines[1])
+	return queue1Cpus, queue2Cpus, nil
+}
+
+// fill the disk before testing to reach the maximum read iops and bandwidth
+// TODO: implement this for windows by passing in the \\\\.\\PhysicalDrive1 parameter
+func fillDisk(symlinkRealPath string) error {
+	if runtime.GOOS == "windows" {
+		fmt.Println("fill disk preliminary step not yet implemented for windows")
+	} else {
+		// hard coding the filesize to 500G as that conforms to the docs while giving
+		// sufficiently high performance
+		fillDiskCmdOptions := fillDiskCommonOptions + " --filesize=500G --filename=" + symlinkRealPath
```

Comment:
Why don't we fill the entire FS? Because it takes too long, or something else?

Reply:
right now, the reason is because it takes too long.

After looking at what the PD team does, it might be possible to have a larger batch size and fill a larger file size, though I'll explore that more once I get pd-balanced to reach the performance targets.